homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -80,6 +135,11 @@ async def async_process(
         self, user_input: conversation.ConversationInput
     ) -> conversation.ConversationResult:
         """Process a sentence."""
+        tools: list[dict[str, Any]] | None = None
```

Comment:
Do the tools need to be constructed on every interaction with the conversation entity? How about constructing them in init once and maybe even share them among the multiple conversation entities?

Reply:
It's up to the LLM API objects to decide if they want to cache them. Here we just fetch them.